### ISSUE TITLE
iOS client not able to PUT/POST

### ISSUE THREAD

**Issue by mostlygeek** [mozilla-services/go-syncstorage#108]
From the nginx access logs, it seem sthat

```
"PUT /1.5/51419390/storage/tabs/mjwJzWFWKUDq HTTP/1.1" 415 37 2182 "Firefox-iOS-Sync/5.1 (Firefox" 0.000 0.000 "415" "TLSv1.2" "ECDHE-RSA-AES256-GCM-SHA384"
```

The `415` for PUT happens [here in the code](https://github.com/mozilla-services/go-syncstorage/blob/8d5152bbc5b90fc60f87a89e64e96f15f83a0500/web/syncUserHandler.go#L880-L885). Excerpt: 

``` go
    // accept text/plain from old (broken) clients
    ct := r.Header.Get("Content-Type")
    if ct != "application/json" && ct != "text/plain" && ct != "application/newlines" {
        JSONError(w, "Not acceptable Content-Type", http.StatusUnsupportedMediaType)
        return
}
```

This error: 

```
"POST /1.5/51419390/storage/history HTTP/1.1" 400 1 29230 "Firefox-iOS-Sync/5.0 (Firefox)" 0.070 0.001 "400" "TLSv1.2" "ECDHE-RSA-AES256-GCM-SHA384"
```

Not quite sure what is causing this. Will need better error reporting. Suspect it also being caused by invalid content types or encoding.

**Comment by mostlygeek** [comment]
After deploying code from #109 it looks like the iOS client the logs say:

```
docker[8561]: time="2016-08-23T22:30:42Z" level=warning msg="HTTP Request Problem" cause="n/a" error="Not acceptable Content-Type: application/json;charset=utf-8" http_code=415 method=PUT path="/1.5/51419390/storage/clients/QYsKUoL42bZF" ua="Firefox-iOS-Sync/5.0 (Firefox)"
```

So it's a bug in the ContentType checking, that I don't parse the values correctly. 👊

**Comment by mostlygeek** [comment]
Fixed in #111
